crates/sui-core/src/authority.rs

```diff
@@ -1024,6 +1029,9 @@ impl AuthorityState {
         let (gas_status, input_objects) =
             transaction_input_checker::check_certificate_input(&self.database, certificate).await?;
 
+        let owned_object_refs = input_objects.filter_owned_objects();
+        self.check_owned_locks(&owned_object_refs).await?;
```

Comment:
We write objects first, then set None locks, which is the source of the race I am fixing here.

We don't strictly need this check here for transactions executed from TransactionManager, because it already verifies this now. But we could still hit the race from the authority server path. I hope to clean up and simplify a lot of this once we are able to remove the sequence tables.